### DISCUSSION TITLE
rootcheck File is owned by root and has written permissions to anyone.

### DISCUSSION THREAD

**Discussion by tootsweets** [wazuh/wazuh#17698] [Q&A]
Hi All

in a test lab, I have a number of hosts that twice a day throw the error 

(level 7) -> "Agent event queue is 90% full."
followed by 
(level 9) -> "Agent event queue is full. Events may be lost."

This happens when the rootcheck scan runs and finds 60000+ File is owned by root and has written permissions to anyone.

I have attempted to resolve this by setting the client buffer to the max and adding a check file no but this never resolved the issue 

	<client_buffer>
		<disabled>no</disabled>
		<queue_size>100000</queue_size>
		<events_per_second>1000</events_per_second>
	</client_buffer>
	<rootcheck>
		<disabled>no</disabled>
		<check_files>no</check_files>
	</rootcheck>
</agent_config>

is it posable to disable this check at the agent level? 

Thanks

**Comment by mdiego92** [comment]
Hello @tootsweets,

Could you please confirm if you have already a `rootcheck` configuration in the `ossec.conf` file from your agent? You can just disable this module by setting `<disable>` to yes.
It should be similar to this one:

```
  <rootcheck>
    <disabled>no</disabled>

    <!-- Frequency that rootcheck is executed - every 12 hours -->
    <frequency>43200</frequency>

    <rootkit_files>/var/ossec/etc/shared/rootkit_files.txt</rootkit_files>
    <rootkit_trojans>/var/ossec/etc/shared/rootkit_trojans.txt</rootkit_trojans>

    <system_audit>/var/ossec/etc/shared/system_audit_rcl.txt</system_audit>
    <system_audit>/var/ossec/etc/shared/system_audit_ssh.txt</system_audit>
    <system_audit>/var/ossec/etc/shared/cis_debian_linux_rcl.txt</system_audit>

    <skip_nfs>yes</skip_nfs>
  </rootcheck>
```

I suppose you have this default configuration because it is being executed every 12 hours (43200 seconds) as set by default in the Agent Installation.

Looking forward to your response!

**Reply by tootsweets** [reply]
Hi

Thanks for getting back to me

yes I have root check in my config the example in my first post is taken from my configuration 

rather than disabling the root check altogether I was hoping the below would work for my issues but it does not

```
<rootcheck>
	<disabled>no</disabled>
	<check_files>no</check_files>
</rootcheck>
```

**Reply by mdiego92** [reply]
Hello @tootsweets,

Could you please share with us the `ossec.log` file from the agent after `rootcheck` execution and flooding?

Another thing that could be useful to analyze the issue is enabling debug mode. For the agent to start in debug mode you should edit the `/var/ossec/etc/local_internal_options.conf` file and set `agent.debug` option to 2. Please restart the agent after these changes and send us the `/var/ossec/logs/ossec.log` file after the `rootcheck` execution.

After getting the logs from the `rootcheck` execution, please revert the changes in the `local_internal_options.conf` file and restart the agent once again.

Looking forward to your response!
